cfme/cloud/instance/__init__.py

```diff
@@ -456,8 +456,8 @@ def all(self):
         # Pretty much same as image, but defining at VMCollection would only work for cloud
         # provider filter means we're viewing instances through provider details relationships
         provider = self.filters.get('provider')  # None if no filter, need for entity instantiation
-        view = navigate_to(self,
-                           'AllForProvider' if provider else 'All')
+        view = navigate_to(provider or self,
+                          'Instances' if provider else 'All')
         # iterate pages here instead of use surf_pages=True because data is needed
         entities = []
         for _ in view.entities.paginator.pages():  # auto-resets to first page
```

Comment:
Definitely want to use `pages()` here, as it resets to the first page and iterates through them - without this, you _could_ be starting on any page.

Reply:
Then I might have to re-fix my test case :roll_eyes: I was facing an issue where `paginator.pages()` was returning nothing since `view.paginator.exists` was returning `False`(check [here](https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L1664)) even though the page was visible in the view.